espnet/st/pytorch_backend/st.py

```diff
@@ -0,0 +1,493 @@
+
+"""Training/decoding definition for RNN sequence-to-sequence speech translation model."""
+
+import json
+import logging
+import os
+import sys
+
+from chainer.datasets import TransformDataset
+from chainer import training
+from chainer.training import extensions
+import numpy as np
+from tensorboardX import SummaryWriter
+import torch
+
+from espnet.asr.asr_utils import adadelta_eps_decay
+from espnet.asr.asr_utils import add_results_to_json
+from espnet.asr.asr_utils import CompareValueTrigger
+from espnet.asr.asr_utils import get_model_conf
+from espnet.asr.asr_utils import restore_snapshot
+from espnet.asr.asr_utils import snapshot_object
+from espnet.asr.asr_utils import torch_load
+from espnet.asr.asr_utils import torch_resume
+from espnet.asr.asr_utils import torch_snapshot
+import espnet.lm.pytorch_backend.extlm as extlm_pytorch
+from espnet.nets.asr_interface import ASRInterface
+from espnet.nets.mt_interface import MTInterface
+from espnet.nets.pytorch_backend.e2e_asr import pad_list
+import espnet.nets.pytorch_backend.lm.default as lm_pytorch
+# from espnet.nets.st_interface import STInterface
+from espnet.utils.deterministic_utils import set_deterministic_pytorch
+from espnet.utils.dynamic_import import dynamic_import
+from espnet.utils.io_utils import LoadInputsAndTargets
+from espnet.utils.training.batchfy import make_batchset
+from espnet.utils.training.iterators import ShufflingEnabler
+from espnet.utils.training.iterators import ToggleableShufflingMultiprocessIterator
+from espnet.utils.training.iterators import ToggleableShufflingSerialIterator
+from espnet.utils.training.tensorboard_logger import TensorboardLogger
+from espnet.utils.training.train_utils import check_early_stop
+from espnet.utils.training.train_utils import set_early_stop
+
+from espnet.asr.pytorch_backend.asr import CustomConverter
+from espnet.asr.pytorch_backend.asr import CustomEvaluator
+from espnet.asr.pytorch_backend.asr import CustomUpdater
+from espnet.asr.pytorch_backend.asr import load_trained_model
+
+import matplotlib
+matplotlib.use('Agg')
+
+if sys.version_info[0] == 2:
+    from itertools import izip_longest as zip_longest
+else:
+    from itertools import zip_longest as zip_longest
+
+
+class CustomConverter_v2(CustomConverter):
+    """Custom batch converter for Pytorch.
+
+    Args:
+        subsampling_factor (int): The subsampling factor.
+        dtype (torch.dtype): Data type to convert.
+        asr_task (bool): multi-task with ASR task.
+
+    """
+
+    def __init__(self, subsampling_factor=1, dtype=torch.float32, asr_task=False):
+        """Construct a CustomConverter_v2 object.
+
+        Args:
+            subsampling_factor (int): The subsampling factor.
+            dtype (torch.dtype): Data type to convert.
+            asr_task (bool): multi-task with ASR task.
+
+        """
+        self.subsampling_factor = subsampling_factor
+        self.ignore_id = -1
+        self.dtype = dtype
+        self.asr_task = asr_task
```

Comment:
```python
super().__init__(subsampling_factor=subsampling_factor, dtype=dtype)
self.asr_task = asr_task
```
is better. It can correctly initialize CustomConverter when it will be updated.